### DISCUSSION TITLE
Inconsistent Naming for Graph Input Types

### DISCUSSION THREAD

**Discussion by MiniDigger** [directus/directus#5058] [Archive]
rc57, pg12, docker

the naming for types between /graphql and /graphql/system is incosistent:

```
update_news_item(data: update_news_input!, id: ID!): news
update_news_items(data: update_news_input!, filter: news_filter, ids: [ID]!, limit: Int, offset: Int, page: Int, search: String, sort: [String]): [news]

vs

update_files_item(data: update_directus_files_input!, id: ID!): directus_files
update_files_items(data: update_directus_files_input!, filter: directus_files_filter, ids: [ID]!, limit: Int, offset: Int, page: Int, search: String, sort: [String]): [directus_files]
```

notice how in system tables, the mutations don't have the directus prefix, but the inputs and the filters have it.

is this expected? I guess the reason is that the inputs and filters are reused in the "non system" schema, but this makes the naming even more confusing...

**Comment by rijkvanzanten** [comment]
This is by design, as the mutation names !== the types they operate on. For example, the system mutation `import_file` operates on the directus_files collection and type. 

Down to discuss potential improvements though 🙂
